src/librustc_resolve/late.rs

```diff
@@ -368,6 +368,19 @@ struct DiagnosticMetadata<'ast> {
     current_let_binding: Option<(Span, Option<Span>, Option<Span>)>,
 }
 
+/// Keeps track of whether errors should be reported.
+///
+/// Used by rustdoc to ignore errors in function bodies.
+#[derive(Copy, Clone, Debug)]
+enum IgnoreState {
+    /// We are at global scope and therefore all errors should be reported.
+    Global,
+    /// We are at function scope and therefore only errors from nested trait impls should be reported.
+    Function,
```

Comment:
These states are redundant, this whole struct can be collapsed into a single boolean.